File: 백준/Silver/13305. 주유소/주유소.py
n = int(input())
road_length = list(map(int, input().split()))
oil_price = list(map(int, input().split()))

cost = 0
min_cost = 2147000000

for i in range(n-1) : 
    if oil_price[i] <= min_cost :                    #최소값 갱신하면서 기름 넣어주기
        cost += oil_price[i] * road_length[i]       #비용 += 오일 가격 * 거리
        min_cost = oil_price[i]                     #최소값 갱신
        continue
    if oil_price[i] > min_cost :                    #최소값보다 오일 가격비 이비싸면 
        cost += min_cost * road_length[i]           #최소값으로 거리 곱해줌

print(cost)

File: 백준/Silver/13305. 주유소/주유소.js
let isBaekjoon = true;
let fs = require("fs");
let input = isBaekjoon
    ? fs.readFileSync("/dev/stdin").toString().split("\n")
    : fs.readFileSync("./test.txt").toString().split("\n");

let n = Number(input[0]);
const loadList = input[1].split(" ").map((el) => Number(el));
const oilList = input[2].split(" ").map((el) => Number(el));

let minCost = 10001;
let sumCost = 0;
for (let i = 0; i < n - 1; i++) {
    //현재 도시를 기준으로 minCost 보다 작으면 그 도시에서 기름 넣고 가자
    if (minCost >= oilList[i]) {
        minCost = oilList[i]; //최소값 갱신
        sumCost += oilList[i] * loadList[i];
    }
    //현재 도시를 기준으로 minCost 보다 큰 경우 minCost 도시에서 기름 넣고 가는걸로 치자
    else {
        sumCost += minCost * loadList[i];
    }
}

console.log(sumCost);